consensus/istanbul/backend/announce.go

```diff
@@ -159,19 +92,28 @@ func (sb *Backend) sendAnnounceMsgs() {
 	}
 }
 
-func (sb *Backend) generateIstAnnounce() ([]byte, error) {
+func (sb *Backend) generateIstAnnounce() (*istanbul.Message, error) {
 	block := sb.currentBlock()
-	selfEnode := sb.Enode()
 
-	if selfEnode == nil {
-		sb.logger.Error("Enode is nil in sendIstAnnounce")
-		return nil, nil
-	}
+	var enodeUrl string
```

Comment:
It's only 50 lines :).

In all seriousness, @mcortesi (the coding style enforcer) has created a function to generate the regAndActiveValidator set in the val_enode_table cleanup PR.  One that PR and this one is merged, it will shorten this function significantly.

Reply:
My bad

Reply:
hahahah.. i like that title... "The Coding style enforcer".... Code Police!